src/lib/protocols/connections/domain/did/__tests__/Service.test.ts

```diff
@@ -0,0 +1,148 @@
+        priority: 10,
+        serviceEndpoint: 'https://example.com',
+      };
+
+      const service = new IndyAgentService({
+        ...json,
+      });
+
+      const transformed = classToPlain(service);
+
+      expect(transformed).toEqual(json);
+    });
+
+    it("should set 'priority' to default (0) when not present in constructor or during transformation", async () => {
+      const json = {
+        id: 'test-id',
+        type: 'IndyAgent',
+        recipientKeys: ['917a109d-eae3-42bc-9436-b02426d3ce2c', '348d5200-0f8f-42cc-aad9-61e0d082a674'],
+        routingKeys: ['0094df0b-7b6d-4ebb-82de-234a621fb615'],
+        serviceEndpoint: 'https://example.com',
+      };
+
+      const transformService = plainToClass(IndyAgentService, json);
+      const constructorService = new IndyAgentService({ ...json });
+
+      expect(transformService.priority).toBe(0);
+      expect(constructorService.priority).toBe(0);
+
+      expect(classToPlain(transformService).priority).toBe(0);
+      expect(classToPlain(constructorService).priority).toBe(0);
+    });
+  });
+
+  describe('ServiceTransformer', () => {
+    class ServiceTransformerTest {
+      @ServiceTransformer()
+      public service: Service[] = [];
+    }
+
+    it("should transform Json to default Service class when the 'type' key is not present in 'serviceTypes'", async () => {
+      const serviceJson = {
+        id: 'test-id',
+        type: 'Mediator',
+        serviceEndpoint: 'https://example.com',
+      };
+
+      const serviceWrapperJson = {
+        service: [serviceJson],
+      };
+      const serviceWrapper = plainToClass(ServiceTransformerTest, serviceWrapperJson);
+
+      expect(serviceWrapper.service.length).toBe(1);
+
+      const firstService = serviceWrapper.service[0];
+      expect(firstService).toBeInstanceOf(Service);
+      expect(firstService.id).toBe(serviceJson.id);
+      expect(firstService.type).toBe(serviceJson.type);
+      expect(firstService.serviceEndpoint).toBe(serviceJson.serviceEndpoint);
+    });
+
+    it("should transform Json to corresponding class when the 'type' key is present in 'serviceTypes'", async () => {
+      const serviceArray = [
+        {
+          id: 'test-id',
+          type: 'IndyAgent',
+          recipientKeys: ['917a109d-eae3-42bc-9436-b02426d3ce2c', '348d5200-0f8f-42cc-aad9-61e0d082a674'],
+          routingKeys: ['0094df0b-7b6d-4ebb-82de-234a621fb615'],
+          priority: 10,
+          serviceEndpoint: 'https://example.com',
+        },
+      ];
+
+      const serviceWrapperJson = {
+        service: serviceArray,
+      };
+      const serviceWrapper = plainToClass(ServiceTransformerTest, serviceWrapperJson);
+
+      expect(serviceWrapper.service.length).toBe(serviceArray.length);
+
+      for (let i = 0; i < serviceArray.length; i++) {
+        const serviceJson = serviceArray[i];
+        const service = serviceWrapper.service[i];
+
+        expect(service).toBeInstanceOf(serviceTypes[serviceJson.type]);
```

Comment:
I usually try to avoid classic `for` loops if possible. Using `forEach` for example seems more readable to me. 

What do you think?
```
serviceArray.forEach((serviceJson, i) => { 
  const service = serviceWrapper.service[i];  
  expect(service).toBeInstanceOf(serviceTypes[serviceJson.type])
})
```